utils/utils.py

```diff
@@ -255,39 +256,57 @@ def load_data_modin_on_omnisci(
+class S3Client:
+    s3_aws_com = ".s3.amazonaws.com"
 
-    data_files_names = list(braceexpand(filename))
+    def __init__(self):
+        self.fs = s3fs.S3FileSystem(anon=True)
 
-    _is_remote_dataset = "://" in filename
-    if _is_remote_dataset:
-        import s3fs
+    @classmethod
+    def s3like(cls, filename: str):
+        if filename.startswith("s3://"):
+            return True
+        elif filename.startswith("https://"):
+            if cls.s3_aws_com not in filename:
+                return False
+            return True
+        else:
+            return False
 
-        fs = s3fs.S3FileSystem(anon=True)
+    def _prepare_s3_link(self, https_link: str):
+        filename = https_link.replace("https://", "")
+        filename = filename.replace(self.s3_aws_com, "")
+        bucket_name = filename.split("/")[0]
+        return bucket_name, filename
 
+    def getsize(self, filename: str):
         if filename.startswith("https://"):
-            s3_aws_com = ".s3.amazonaws.com"
-            if s3_aws_com not in filename:
-                raise ValueError(
-                    "https format of S3 links supported only for aws; bad https link: {filename}"
-                )
+            _, s3_filename = self._prepare_s3_link(filename)
+        return self.fs.info(s3_filename)["Size"]
+
+    def glob(self, files_pattern: str):
+        if files_pattern.startswith("https://"):
+            bucket_name, s3_files_pattern = self._prepare_s3_link(files_pattern)
+            return [
+                f"https://{filename.replace(bucket_name, bucket_name+self.s3_aws_com)}"
+                for filename in self.fs.glob(s3_files_pattern)
+            ]
+        else:
+            return [f"s3://{filename}" for filename in self.fs.glob(files_pattern)]
 
-            def http_glob(filename):
-                new_filename = filename.replace("https://", "")
-                new_filename = new_filename.replace(s3_aws_com, "")
-                bucket_name = new_filename.split("/")[0]
-                return [
-                    f"https://{x.replace(bucket_name, bucket_name+s3_aws_com)}"
-                    for x in fs.glob(new_filename)
-                ]
 
-            data_files_names = sorted([x for f in data_files_names for x in http_glob(f)])
+s3_client = S3Client()
 
-        elif filename.startswith("s3://"):
-            data_files_names = sorted([f"s3://{x}" for f in data_files_names for x in fs.glob(f)])
-        else:
-            raise ValueError(f"bad filename: '{filename}'; prefix should be 'https://' or 's3://'")
+
+def files_names_from_pattern(files_pattern):
+    from braceexpand import braceexpand
+
+    data_files_names = list(braceexpand(files_pattern))
+
+    if "://" in files_pattern:
+        if all(map(s3_client.s3like, data_files_names)):
+            data_files_names = sorted([x for f in data_files_names for x in s3_client.glob(f)])
+        raise ValueError(f"some of s3like links are bad: {data_files_names}")
```

Comment:
```suggestion
        else:
            raise ValueError(f"some of s3like links are bad: {data_files_names}")
```
hmm... did you check your last changes? :)